multinode_integration_tests/tests/verify_bill_payment_utils/utils.rs

```diff
@@ -0,0 +1,1003 @@
+    ))
+}
+
+fn load_contract_in_bytes() -> Vec<u8> {
+    let file_path =
+        standard_dir_for_test_input_data().join("smart_contract_for_on_blockchain_test");
+    let mut file = File::open(file_path).expect("couldn't acquire a handle to the data file");
+    let mut data = String::new();
+    file.read_to_string(&mut data).unwrap();
+    let data = data
+        .chars()
+        .filter(|char| !char.is_whitespace())
+        .collect::<String>();
+    data.from_hex::<Vec<u8>>()
+        .expect("bad contract: contains non-hexadecimal characters")
+}
+
+lazy_static! {
+    static ref GAS_PRICE: ethereum_types::U256 =
+        50_u64.try_into().expect("Gas price, internal error");
+    static ref GAS_LIMIT: ethereum_types::U256 =
+        1_000_000_u64.try_into().expect("Gas limit, internal error");
+}
+
+fn deploy_smart_contract(wallet: &Wallet, web3: &Web3<Http>, chain: Chain) -> Address {
+    let contract = load_contract_in_bytes();
+    let tx = TransactionParameters {
+        nonce: Some(ethereum_types::U256::zero()),
+        to: None,
+        gas: *GAS_LIMIT,
+        gas_price: Some(*GAS_PRICE),
+        value: ethereum_types::U256::zero(),
+        data: Bytes(contract),
+        chain_id: Some(chain.rec().num_chain_id),
+    };
+    let signed_tx = primitive_sign_transaction(web3, tx, wallet);
+    match web3
+        .eth()
+        .send_raw_transaction(signed_tx.raw_transaction)
+        .wait()
+    {
+        Ok(tx_hash) => match web3.eth().transaction_receipt(tx_hash).wait() {
+            Ok(Some(tx_receipt)) => tx_receipt.contract_address.unwrap(),
+            Ok(None) => panic!("Contract deployment failed Ok(None)"),
+            Err(e) => panic!("Contract deployment failed {:?}", e),
+        },
+        Err(e) => panic!("Contract deployment failed {:?}", e),
+    }
+}
+
+fn transfer_service_fee_amount_to_address(
+    contract_addr: Address,
+    from_wallet: &Wallet,
+    to_wallet: &Wallet,
+    amount_minor: u128,
+    transaction_nonce: u64,
+    web3: &Web3<Http>,
+    chain: Chain,
+) {
+    let data = transaction_data_web3(to_wallet, amount_minor);
+    let tx = TransactionParameters {
+        nonce: Some(ethereum_types::U256::try_from(transaction_nonce).expect("Internal error")),
+        to: Some(contract_addr),
+        gas: *GAS_LIMIT,
+        gas_price: Some(*GAS_PRICE),
+        value: ethereum_types::U256::zero(),
+        data: Bytes(data.to_vec()),
+        chain_id: Some(chain.rec().num_chain_id),
+    };
+    let signed_tx = primitive_sign_transaction(web3, tx, from_wallet);
+    match web3
+        .eth()
+        .send_raw_transaction(signed_tx.raw_transaction)
+        .wait()
+    {
+        Ok(tx_hash) => eprintln!(
+            "Transaction {:?} of {} wei of MASQ was sent from wallet {} to {}",
+            tx_hash, amount_minor, from_wallet, to_wallet
+        ),
+        Err(e) => panic!("Transaction for token transfer failed {:?}", e),
+    }
+}
+
+fn primitive_sign_transaction(
```

Comment:
At the time of reviewing, you were seriously pissed off with the word `primitive`, you don't want it all.